### PR TITLE
refactor: add OpenAPI schema to simple-password-provider controller

### PR DESCRIPTION
**Pull request by nunogois**
https://trello.com/c/x8kbp2dJ

Waiting for https://github.com/Unleash/unleash/pull/1692 to be merged first.

### CODE REVIEW

**Review by andreas-unleash — APPROVED** [review]

File: src/lib/openapi/index.ts [hunk 1]
```diff
@@ -178,6 +180,12 @@ export type AdminApiOperation = ApiOperation<'admin'>;
 export type ClientApiOperation = ApiOperation<'client'>;
 export type OtherApiOperation = ApiOperation<'other'>;
 
+export interface AuthApiOperation
```
Comment (olav) [inline src/lib/openapi/index.ts:183]: Could be declared the same way as `OtherApiOperation` etc above.

File: src/lib/routes/auth/simple-password-provider.test.ts [hunk 2]
```diff
@@ -1,18 +1,24 @@
...
+    const openApiService = new OpenApiService(config);
     const app = express();
     app.use(express.json());
     const userService = () => {};
-    // @ts-ignore
-    const ctr = new PasswordProvider({ getLogger }, { userService });
 
-    //@ts-ignore
+    const ctr = new SimplePasswordProvider(config, {
+        // @ts-expect-error
```
Comment (olav) [inline src/lib/routes/auth/simple-password-provider.test.ts:17]: Nice!

File: src/lib/services/openapi-service.ts [hunk 3]
```diff
@@ -32,7 +33,11 @@ export class OpenApiService {
     }
 
     validPath(
-        op: AdminApiOperation | ClientApiOperation | OtherApiOperation,
+        op:
+            | AdminApiOperation
```
Comment (olav) [inline src/lib/services/openapi-service.ts:37]: Could refactor this while we're here:

```ts
validPath(op: ApiOperation): RequestHandler {
```

**Review by olav — CHANGES_REQUESTED** [review]

File: src/lib/services/openapi-service.ts (hunk 3 above)
Comment (nunogois) [reply src/lib/services/openapi-service.ts]: Do you mean like this? https://github.com/Unleash/unleash/pull/1734/commits/cb0efcc0096d4b264df3afea19462e71f67fbbde

Comment (olav) [reply src/lib/services/openapi-service.ts]: Not exactly, but your way seems even better. :tada:

**Review by olav — APPROVED** [review]

**Pull request merged**